packages/caliper-core/test/common/core/transaction-statisitcs-collector.js

```diff
@@ -0,0 +1,219 @@
+    let clock;
+
+    beforeEach(function() {
+        sandbox = sinon.createSandbox();
+        clock = sandbox.useFakeTimers();
+    });
+
+    afterEach(function() {
+        sandbox.restore();
+    });
+
+    describe('Using a single TransactionStatisticsCollector to collect results', () => {
+
+        it('should not be possible to get basic round information from the collector', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector(0, 1, 'roundLabel');
+
+            transactionStatisticsCollector.getWorkerIndex().should.equal(0);
+            transactionStatisticsCollector.getRoundIndex().should.equal(1);
+            transactionStatisticsCollector.getRoundLabel().should.equal('roundLabel');
+        });
+
+        it('should not be possible to update a TransactionStatisticsCollector if it is not activated', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+
+            const updateSpy = sandbox.spy(transactionStatisticsCollector, '_updateStatistics');
+
+            transactionStatisticsCollector.txSubmitted(1);
+            transactionStatisticsCollector.getTotalSubmittedTx().should.equal(0);
+
+            const txStatus = new TxStatus();
+            transactionStatisticsCollector.txFinished(txStatus);
+            sinon.assert.notCalled(updateSpy);
+        });
+
+        it('should be possible to update a TransactionStatisticsCollector with a single TxStatus results once activated', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            const updateSpy = sandbox.spy(transactionStatisticsCollector, '_updateStatistics');
+
+            transactionStatisticsCollector.txSubmitted(1);
+            transactionStatisticsCollector.getTotalSubmittedTx().should.equal(1);
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            transactionStatisticsCollector.txFinished(txStatus);
+            sinon.assert.calledOnce(updateSpy);
+        });
+
+        it('should be possible to update a TransactionStatisticsCollector with an array of TxStatus results once activated', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            const updateSpy = sandbox.spy(transactionStatisticsCollector, '_updateStatistics');
+
+            transactionStatisticsCollector.txSubmitted(1);
+            transactionStatisticsCollector.getTotalSubmittedTx().should.equal(1);
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            transactionStatisticsCollector.txFinished([txStatus, txStatus]);
+            sinon.assert.calledTwice(updateSpy);
+        });
+
+        it('should not be possible to update a TransactionStatisticsCollector with a TxStatus result once deactivated', () => {
```

Comment:
The active state is already tested above. So only the deactivated state should be tested, and separately for submitted and finished TXs.